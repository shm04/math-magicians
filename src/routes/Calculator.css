.calculator-div {
  display: flex;
}

.calculator-title {
  font-size: 30px;
}
