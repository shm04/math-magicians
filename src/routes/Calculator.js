import Calculator from '../components/calculator';
import './Calculator.css';

function CalculatorDiv() {
  return (
    <div className="calculator-div">
      <h3 className="calculator-title">LetÂ´s do some math!</h3>
      <Calculator />
    </div>
  );
}

export default CalculatorDiv;
