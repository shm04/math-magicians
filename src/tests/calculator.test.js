import { render } from '@testing-library/react';
import Calculator from '../components/calculator';

describe('Calculator', () => {
  it('Render calculator', () => {
    const { container } = render(<Calculator />);
    expect(container).toMatchSnapshot();
  });
});
