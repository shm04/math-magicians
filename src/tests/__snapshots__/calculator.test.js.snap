// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator Render calculator 1`] = `
<div>
  <div
    class="main-div"
  >
    <p
      class="calc-screen"
    >
      0
    </p>
    <div
      class="btns-div"
    >
      <button
        class="calc-btn"
        type="button"
      >
        AC
      </button>
      <button
        class="calc-btn"
        type="button"
      >
        +/-
      </button>
      <button
        class="calc-btn"
        type="button"
      >
        %
      </button>
      <button
        class="calc-btn orange-btns"
        type="button"
      >
        รท
      </button>
      <button
        class="calc-btn"
        type="button"
      >
        7
      </button>
      <button
        class="calc-btn"
        type="button"
      >
        8
      </button>
      <button
        class="calc-btn"
        type="button"
      >
        9
      </button>
      <button
        class="calc-btn orange-btns"
        type="button"
      >
        x
      </button>
      <button
        class="calc-btn"
        type="button"
      >
        4
      </button>
      <button
        class="calc-btn"
        type="button"
      >
        5
      </button>
      <button
        class="calc-btn"
        type="button"
      >
        6
      </button>
      <button
        class="calc-btn orange-btns"
        type="button"
      >
        -
      </button>
      <button
        class="calc-btn"
        type="button"
      >
        1
      </button>
      <button
        class="calc-btn"
        type="button"
      >
        2
      </button>
      <button
        class="calc-btn"
        type="button"
      >
        3
      </button>
      <button
        class="calc-btn orange-btns"
        type="button"
      >
        +
      </button>
      <button
        class="calc-btn"
        type="button"
      >
        0
      </button>
      <button
        class="calc-btn"
        type="button"
      >
        .
      </button>
      <button
        class="calc-btn orange-btns"
        id="equal-btn"
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`Calculator Renders calculator correctly 1`] = `
<div>
  <div
    class="main-div"
  >
    <p
      class="calc-screen"
    >
      0
    </p>
    <div
      class="btns-div"
    >
      <button
        class="calc-btn"
        type="button"
      >
        AC
      </button>
      <button
        class="calc-btn"
        type="button"
      >
        +/-
      </button>
      <button
        class="calc-btn"
        type="button"
      >
        %
      </button>
      <button
        class="calc-btn orange-btns"
        type="button"
      >
        รท
      </button>
      <button
        class="calc-btn"
        type="button"
      >
        7
      </button>
      <button
        class="calc-btn"
        type="button"
      >
        8
      </button>
      <button
        class="calc-btn"
        type="button"
      >
        9
      </button>
      <button
        class="calc-btn orange-btns"
        type="button"
      >
        x
      </button>
      <button
        class="calc-btn"
        type="button"
      >
        4
      </button>
      <button
        class="calc-btn"
        type="button"
      >
        5
      </button>
      <button
        class="calc-btn"
        type="button"
      >
        6
      </button>
      <button
        class="calc-btn orange-btns"
        type="button"
      >
        -
      </button>
      <button
        class="calc-btn"
        type="button"
      >
        1
      </button>
      <button
        class="calc-btn"
        type="button"
      >
        2
      </button>
      <button
        class="calc-btn"
        type="button"
      >
        3
      </button>
      <button
        class="calc-btn orange-btns"
        type="button"
      >
        +
      </button>
      <button
        class="calc-btn"
        type="button"
      >
        0
      </button>
      <button
        class="calc-btn"
        type="button"
      >
        .
      </button>
      <button
        class="calc-btn orange-btns"
        id="equal-btn"
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
